syntax="proto3";
package replication;

enum synctype{
  NF = 0;
  FULLSYNC = 1;
  PARTSYNC = 2;
}

message command_slaveof{
  string ip = 1;
  int64 port = 2;
}

// ======================== repl ping pong ==============================
message command_ping{
  string ip = 1;
}

message command_pong{
  string ip = 1;
}

// ======================== repl sync ==============================
message command_ack{
  string repl_aof_file_name = 1;
  uint64 repl_aof_off = 2;
}

message command_psync{
  string run_id = 1;
  string repl_aof_file_name = 2;
  uint64 repl_aof_off = 3;
}

message command_psync_reply{
  int32 errcode = 1;
  string errmsg = 2;
  synctype psync_type = 3; // 1-全量同步 2-部分同步
  string run_id = 4;
  int64 repl_database_size = 5; // 全量同步字段
  // 当全量同步时，值为dump database时之后的增量位置
  // 当部分同步时，值为已经同步完成的增量位置
  string repl_aof_file_name = 6;
  int64 repl_aof_off = 7;
}